### PR TITLE
Optimized Tests

### PR DESCRIPTION
**Pull request by christopher-besch**
# General Problem

Manim testing is very slow and should be optimized.

You can get the slowest 50 test with:
```
pytest --durations=50
```
Please run this command and post its output here. You should also state which OS you're using since—as you'll see in a moment—there are huge differences between them.

This is the output on a macbook (from @behackl):
```
133.24s call     tests/test_scene_rendering/test_cli_flags.py::test_resolution_flag
123.73s call     tests/test_scene_rendering/opengl/test_cli_flags_opengl.py::test_resolution_flag
72.39s call     tests/test_plugins/test_plugins.py::test_plugin_function_like
43.44s call     tests/test_graphical_units/test_threed.py::test_SurfaceColorscale
40.67s call     tests/test_plugins/test_plugins.py::test_plugin_no_all
39.50s call     tests/test_plugins/test_plugins.py::test_plugin_with_all
26.57s call     tests/test_graphical_units/test_threed.py::test_Arrow3D
25.82s call     tests/test_scene_rendering/opengl/test_caching_related_opengl.py::test_play_skip
19.50s call     tests/test_graphical_units/test_threed.py::test_Line3D
17.56s call     tests/test_scene_rendering/opengl/test_caching_related_opengl.py::test_wait_skip
16.73s call     tests/test_scene_rendering/opengl/test_cli_flags_opengl.py::test_n_flag
16.61s call     tests/test_scene_rendering/opengl/test_cli_flags_opengl.py::test_basic_scene_with_default_values
16.01s call     tests/test_graphical_units/test_coordinate_systems.py::test_number_plane_log
15.76s call     tests/test_logging/test_logging.py::test_error_logging
14.90s call     tests/test_graphical_units/test_threed.py::test_Cone
14.87s call     tests/test_scene_rendering/test_cli_flags.py::test_basic_scene_with_default_values
14.42s call     tests/test_scene_rendering/test_caching_related.py::test_play_skip
14.04s call     tests/test_coordinate_system.py::test_NumberPlane
13.47s call     tests/opengl/test_texmobject_opengl.py::test_tex_whitespace_arg
13.35s call     tests/test_scene_rendering/test_cli_flags.py::test_n_flag
12.80s call     tests/opengl/test_texmobject_opengl.py::test_tex_white_space_and_non_whitespace_args
12.45s call     tests/opengl/test_texmobject_opengl.py::test_double_braces_testing[{{ a }} + {{ b }} = {{ c }}-5]
12.23s call     tests/opengl/test_texmobject_opengl.py::test_tex
12.19s call     tests/test_config.py::test_transparent
12.06s call     tests/test_scene_rendering/test_caching_related.py::test_wait_skip
11.80s call     tests/opengl/test_texmobject_opengl.py::test_tex_non_whitespace_arg
11.61s call     tests/test_scene_rendering/test_sections.py::test_no_sections
11.34s call     tests/test_graphical_units/test_axes.py::test_plot
11.23s call     tests/test_graphical_units/test_numbers.py::test_set_value_with_updaters
10.93s call     tests/test_scene_rendering/test_sections.py::test_sections
10.78s call     tests/test_scene_rendering/test_sections.py::test_many_sections
10.13s call     tests/test_graphical_units/test_tables.py::test_Table
10.01s call     tests/test_logging/test_logging.py::test_logging_to_file
```

This is how it looks like (without the changes of this PR) on my machine—Fedora Linux:
```
35.93s call     tests/test_scene_rendering/test_cli_flags.py::test_resolution_flag
33.12s call     tests/test_scene_rendering/opengl/test_cli_flags_opengl.py::test_resolution_flag
16.47s call     tests/test_graphical_units/test_threed.py::test_SurfaceColorscale
10.18s call     tests/test_plugins/test_plugins.py::test_plugin_function_like
9.66s call     tests/test_plugins/test_plugins.py::test_plugin_no_all
8.42s call     tests/test_plugins/test_plugins.py::test_plugin_with_all
7.69s call     tests/test_graphical_units/test_coordinate_systems.py::test_number_plane_log
7.63s call     tests/test_scene_rendering/test_cli_flags.py::test_n_flag
7.57s call     tests/test_scene_rendering/opengl/test_cli_flags_opengl.py::test_n_flag
6.16s call     tests/test_coordinate_system.py::test_NumberPlane
5.62s call     tests/opengl/test_texmobject_opengl.py::test_double_braces_testing[{{ a }} + {{ b }} = {{ c }}-5]
5.12s call     tests/test_graphical_units/test_tables.py::test_Table
5.10s call     tests/test_scene_rendering/opengl/test_caching_related_opengl.py::test_play_skip
5.09s call     tests/test_texmobject.py::test_tex_whitespace_arg
4.91s call     tests/test_graphical_units/test_coordinate_systems.py::test_number_plane
4.85s call     tests/test_scene_rendering/test_caching_related.py::test_play_skip
4.80s call     tests/opengl/test_texmobject_opengl.py::test_tex_non_whitespace_arg
4.63s call     tests/opengl/test_texmobject_opengl.py::test_tex_white_space_and_non_whitespace_args
4.61s call     tests/test_texmobject.py::test_double_braces_testing[{{ a }} + {{ b }} = {{ c }}-5]
4.55s call     tests/test_graphical_units/test_threed.py::test_Arrow3D
4.50s call     tests/test_number_line.py::test_add_labels
4.45s call     tests/opengl/test_texmobject_opengl.py::test_tex_whitespace_arg
4.28s call     tests/test_scene_rendering/test_sections.py::test_many_sections
4.17s call     tests/opengl/test_number_line_opengl.py::test_add_labels
4.12s call     tests/test_graphical_units/test_threed.py::test_Line3D
4.06s call     tests/test_scene_rendering/test_sections.py::test_skip_animations
4.03s call     tests/test_graphical_units/test_axes.py::test_log_scaling_graph
3.81s call     tests/test_texmobject.py::test_tex_non_whitespace_arg
3.78s call     tests/test_graphical_units/test_axes.py::test_plot_line_graph
3.63s call     tests/test_scene_rendering/opengl/test_caching_related_opengl.py::test_wait_skip
3.55s call     tests/test_graphical_units/test_vector_scene.py::test_vector_to_coords
3.54s call     tests/test_texmobject.py::test_tex_white_space_and_non_whitespace_args
3.41s call     tests/test_scene_rendering/test_cli_flags.py::test_basic_scene_with_default_values
3.38s call     tests/test_scene_rendering/opengl/test_cli_flags_opengl.py::test_basic_scene_with_default_values
3.27s call     tests/test_scene_rendering/test_caching_related.py::test_wait_skip
3.19s call     tests/test_graphical_units/test_numbers.py::test_set_value_with_updaters
3.00s call     tests/test_scene_rendering/opengl/test_opengl_renderer.py::test_force_window_opengl_render_with_format
2.87s call     tests/test_graphical_units/test_threed.py::test_Cone
```

To get the time required for the entire test suit, you can use the GNU time tool like this:
```
\time pytest
```
Make sure to add the `\` as you have to use the GNU time binary and not the shell builtin. You might not need this if you're using a very exotic shell.

You'll receive an output similar to this:
```
487.73user 269.32system 2:05.35elapsed 603%CPU (0avgtext+0avgdata 802116maxresident)k
0inputs+39152outputs (184major+11896649minor)pagefaults 0swaps
```
The value you're looking for is `xx:yy.zzelapsed`. This is the total amount of wall time the user has to wait for the program to finish.

# Effects of this PR

This PR is a first attempt at that and is by no means exhaustive. It optimizes a few test while attempting to remove as little testing coverage as possible.
This is my new output:
```
17.75s call     tests/test_graphical_units/test_threed.py::test_SurfaceColorscale
10.50s call     tests/test_plugins/test_plugins.py::test_plugin_function_like
9.93s call     tests/test_plugins/test_plugins.py::test_plugin_no_all
8.90s call     tests/test_plugins/test_plugins.py::test_plugin_with_all
8.82s call     tests/test_scene_rendering/test_cli_flags.py::test_resolution_flag
7.33s call     tests/test_graphical_units/test_coordinate_systems.py::test_number_plane_log
7.32s call     tests/test_scene_rendering/opengl/test_cli_flags_opengl.py::test_resolution_flag
6.34s call     tests/opengl/test_texmobject_opengl.py::test_double_braces_testing[{{ a }} + {{ b }} = {{ c }}-5]
6.05s call     tests/test_coordinate_system.py::test_NumberPlane
5.46s call     tests/test_texmobject.py::test_tex_whitespace_arg
5.44s call     tests/opengl/test_texmobject_opengl.py::test_tex_non_whitespace_arg
5.21s call     tests/opengl/test_texmobject_opengl.py::test_tex_white_space_and_non_whitespace_rgs
5.11s call     tests/test_scene_rendering/opengl/test_caching_related_opengl.py::test_play_skip
5.07s call     tests/test_graphical_units/test_tables.py::test_Table
5.05s call     tests/test_graphical_units/test_threed.py::test_Arrow3D
4.83s call     tests/test_scene_rendering/test_caching_related.py::test_play_skip
4.79s call     tests/opengl/test_number_line_opengl.py::test_add_labels
4.63s call     tests/opengl/test_texmobject_opengl.py::test_tex_whitespace_arg
4.47s call     tests/test_graphical_units/test_threed.py::test_Line3D
4.46s call     tests/test_texmobject.py::test_double_braces_testing[{{ a }} + {{ b }} = {{ c }}-5]
4.43s call     tests/test_number_line.py::test_add_labels
4.42s call     tests/test_graphical_units/test_axes.py::test_log_scaling_graph
4.22s call     tests/test_scene_rendering/test_sections.py::test_many_sections
4.07s call     tests/test_graphical_units/test_functions.py::test_ImplicitFunction
```

With this, the tests take a total of **1min and 40sec** instead of **2min 5sec** without the changes—20% improvement.
This isn't too much, but should be detectable by and just the beginning. It shows how small changes can already improve testing times considerably.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: tests/test_scene_rendering/opengl/test_cli_flags_opengl.py [hunk 1]
```diff
@@ -36,22 +36,21 @@ def test_basic_scene_with_default_values(tmp_path, manim_cfg_file, simple_scenes
...
-        (1920, 1080),
-        (2560, 1440),
-        (3840, 2160),
-        (640, 480),
-        (800, 600),
+        (720, 480, ";"),
+        (1280, 720, ","),
+        (1920, 1080, "-"),
+        (2560, 1440, ";"),
+        # (3840, 2160, ","),
```
Comment (ryanmccauley211) [inline tests/test_scene_rendering/opengl/test_cli_flags_opengl.py:48]: Are these just left commented out to allow for local testing of these resolutions?

**Review by ryanmccauley211 — APPROVED** [review]

**Review by naveen521kk — APPROVED** [review]
LGTM

File: tests/test_scene_rendering/opengl/test_cli_flags_opengl.py (hunk 1 above)
Comment (christopher-besch) [reply tests/test_scene_rendering/opengl/test_cli_flags_opengl.py:48]: Nope, I each one of these tests takes ages and do we really need that many different resolutions to be tested?

Comment (ryanmccauley211) [reply tests/test_scene_rendering/opengl/test_cli_flags_opengl.py:48]: I don’t think so, think we can just remove the commented out ones ?

Comment (christopher-besch) [reply tests/test_scene_rendering/opengl/test_cli_flags_opengl.py:48]: Ups, I misunderstood your first comment. That's exactly why they're commented out and not deleted.

**Review by kolibril13 — APPROVED** [review]

**Pull request merged**